pkg/storage/azureblob.go

```diff
@@ -65,14 +63,21 @@ func (u *AzureBlob) Upload(ldif []byte, log []byte) error {
 	return nil
 }
 
-func (u *AzureBlob) uploadFile(name string, content []byte) error {
-	blobReference := u.Container.GetBlobReference(name)
+func (u *AzureContainer) uploadFile(name string, content []byte) error {
+	filePath := fmt.Sprintf("/tmp/%s", name)
+	err := ioutil.WriteFile(filePath, content, 0700)
+	if err != nil {
+		return err
+	}
 
-	// create the blob if it does not exist
-	err := blobReference.PutAppendBlob(nil)
-	if err == nil {
-		err = blobReference.AppendBlock(content, nil)
+	blobURL := azblob.ContainerURL(*u.Container).NewBlockBlobURL(name)
+	file, err := os.Open(filePath)
+	if err != nil {
+		return err
 	}
 
+	ctx := context.Background()
+	_, err = azblob.UploadFileToBlockBlob(ctx, file, blobURL, azblob.UploadToBlockBlobOptions{})
+
 	return err
 }
```

Comment:
I think we can use the UploadBufferToBlockBlob function to upload the content directly without needing to persist it into a file first.
```go
func (u *AzureContainer) uploadFile(name string, content []byte) error {
	blobURL := azblob.ContainerURL(*u.Container).NewBlockBlobURL(name)
	ctx := context.Background()
	_, err := azblob.UploadBufferToBlockBlob(ctx, content, blobURL, azblob.UploadToBlockBlobOptions{})
	return err
}
```

Reply:
Good point. As `UploadFileToBlockBlob` calls in the end `UploadBufferToBlockBlob`.